main.js

```diff
@@ -1,107 +1,290 @@
+        }
+        // Function to start the confetti animation
+        const startConfetti = () => {
+            if (this.gameoverMessage) {
+                this.gameoverMessage.textContent = "You Won!";
+            }
+            confettiArray.forEach((confetti) => {
+                document.body.appendChild(confetti);
+            });
+        };
+        // Function to stop the confetti animation
+        const stopConfetti = () => {
+            confettiArray.forEach((confetti) => {
+                confetti.remove();
+            });
+        };
+        // Start the confetti animation
+        startConfetti();
+        // Stop the confetti animation after 5 seconds
+        setTimeout(stopConfetti, 5000);
+    }
+}
+class ElementCreator {
+    name;
+    width;
+    height;
+    image;
+    className;
+    style;
+    constructor(name, width, height, image, style, className) {
+        this.name = name;
+        this.width = width;
+        this.height = height;
+        if (image) {
+            this.image = image;
+        }
+        if (style) {
+            this.style = style;
+        }
+        if (className) {
+            this.className = className;
+        }
+    }
+    createDiv() {
+        const div = document.createElement("div");
+        div.style.width = this.width + "px";
+        div.style.height = this.height + "px";
+        if (this.className) {
+            div.classList.add(this.className ? this.className : "");
+        }
+        if (this.style) {
+            Object.assign(div.style, this.style);
+        }
+        return div;
+    }
+    createImg() {
+        const img = document.createElement("img");
+        img.src = this.image ? this.image : "";
+        img.style.width = this.width + "px";
+        img.style.height = this.height + "px";
+        if (this.className) {
+            img.classList.add(this.className ? this.className : "");
+        }
+        if (this.style) {
+            Object.assign(img.style, this.style);
+        }
+        return img;
+    }
+    createElement() {
+        switch (this.name.toUpperCase()) {
+            case "DIV":
+                return this.createDiv();
+            case "IMG":
+                return this.createImg();
+        }
+    }
+}
+class Block {
+    name;
+    width;
+    height;
+    image;
+    imageElement;
+    divElement;
+    openFunction;
+    constructor(name, width, height, image, openFunction) {
         this.name = name;
         this.width = width;
         this.height = height;
         this.image = image;
     }
-    Block.prototype.getAttributes = function () {
+    setOpenFunction(openFunction) {
+        this.openFunction = openFunction;
+    }
+    // Created a single function for opening so that we can later disable the event listener
+    // When disabling an event listener we need to pass the same exact function that was initially passed
+    // Anonymous functions cannot be later disabled
+    getOpenFunction() {
```

Comment:
Why would I need get for openFunction if openFunction is publicly accessible from everywhere?